torch/distributed/_sharded_tensor/api.py

```diff
@@ -335,6 +363,62 @@ def _init_rpc(self):
         # Barrier for all RPCs to finish on all ranks.
         rpc.api._all_gather(None)
 
+    def gather(
+        self,
+        dst: int = 0,
+        out: Optional[torch.Tensor] = None,
+    ) -> None:
+        """
+        Creates a full :class:`Tensor` on rank `dst` by gathering all sharded tensors.
+
+        The API needs to be called on all ranks in SPMD fashion. All ranks should have
+        the same `dst`. `out` should be a full size zero tensor on `dst` and
+        None on all other ranks.
+
+        TODO: current only supports GPU. Will enable CPU and revise to replace
```

Comment:
nit: probably add a warning instead?

Reply:
don't put todo in public docs, put this as a comment in the code and in the comment link to the appropriate gh issue.